src/api/dtoTransformere.ts

```diff
@@ -0,0 +1,74 @@
+import { AxiosResponse } from 'axios';
+import { TiltakInstansDTO } from './data/tiltak';
+import { TiltakInstans, TiltakInstansStatus } from '../domeneobjekter/tiltak';
+import { TiltakDeltagerDetaljerDTO, TiltakDeltagerDTO } from './data/deltager';
+import { TiltakDeltager, TiltakDeltagerDetaljer, TiltakDeltagerStatus } from '../domeneobjekter/deltager';
+import { AnsattRolle, InnloggetAnsatt } from '../domeneobjekter/ansatt';
+import { AnsattRolleDTO, InnloggetAnsattDTO } from './data/ansatt';
+
+export const transformTiltakInstanser = (response: AxiosResponse<TiltakInstansDTO[]>) : AxiosResponse<TiltakInstans[]> => {
+    const tiltak = response.data.map(toTiltakInstans);
+    return {...response, data: tiltak}
+}
+
+export const transformTiltakInstans = (response: AxiosResponse<TiltakInstansDTO>) : AxiosResponse<TiltakInstans> => {
+    return {...response, data: toTiltakInstans(response.data)}
+}
+
+const toTiltakInstans = (tiltakInstansDTO: TiltakInstansDTO) : TiltakInstans => {
+    return {
+        ...tiltakInstansDTO,
+        startdato: new Date(tiltakInstansDTO.startdato),
+        sluttdato: new Date(tiltakInstansDTO.sluttdato),
+        status: TiltakInstansStatus[tiltakInstansDTO.status]
+    };
+}
+
+export const transformTiltakDeltager = (response: AxiosResponse<TiltakDeltagerDTO[]>) : AxiosResponse<TiltakDeltager[]> => {
+    const deltagere =  response.data.map(toTiltakDeltager);
+    return {...response, data: deltagere}
+}
+
+const toTiltakDeltager = (tiltakDeltagerDTO : TiltakDeltagerDTO) : TiltakDeltager => {
+    return {
+        ...tiltakDeltagerDTO,
+        startdato: new Date(tiltakDeltagerDTO.startdato),
+        sluttdato: new Date(tiltakDeltagerDTO.sluttdato),
+        status: TiltakDeltagerStatus[tiltakDeltagerDTO.status]
+    }
+}
+
+export const transformTiltakDeltagerDetaljer = (response: AxiosResponse<TiltakDeltagerDetaljerDTO>) : AxiosResponse<TiltakDeltagerDetaljer> => {
+    return {...response, data: toTiltakDeltagerDetaljer(response.data)};
+}
+
+const toTiltakDeltagerDetaljer = (tiltakDeltagerDetaljerDTO: TiltakDeltagerDetaljerDTO) : TiltakDeltagerDetaljer => {
+    return {
+        ...tiltakDeltagerDetaljerDTO,
+        startdato: new Date(tiltakDeltagerDetaljerDTO.startdato),
```

Comment:
Man kan vurdere å bruke en type istedenfor Date hvis vi uansett skal bruke dayjs for formatering.
F.eks `type LocalDate = string;  type ZonedDateTime = string;` for å skille mellom forskjellige dato typer

Reply:
Vet ikke om jeg forstår helt. Foretrekker at det er typa som Date så det er innlysende hva det er og så kan man fange opp om det ikke er en dato

Reply:
Problemet med Date er at det ikke sier noe om at det er et timestamp eller en dato (uten tid). Date gir heller ikke så mye verdi for resten av koden siden vi uansett må konvertere til en dayjs dato.

Reply:
Hvis tanken bak denne PRen er å fange opp feil typer, så mangler det validering av responsen. Typescript gir ingen garantier på at typene etter man har brukt transform() funksjonene faktisk er riktig